scripts/tools/zap/generate.py

```diff
@@ -83,18 +86,23 @@ def detectZclFile(zapFile):
     path = 'src/app/zap-templates/zcl/zcl.json'
 
     data = json.load(open(zapFile))
+    prefix_chip_root_dir = True
     for package in data["package"]:
         if package["type"] != "zcl-properties":
             continue
 
+        prefix_chip_root_dir = True
```

Comment:
why is this set to true in a loop? Why is the set to true at the top not sufficient?

Reply:
maybe instead of looping, since the path is fetched in a loop, the right thing to do is:

`prefix_chip_root_dir = (package["pathRelativity"] != "resolveEnvVars")`

Reply:
One can have multiple packages with type==zcl-properties in the zap file, I think zap will load all of them (but I'm NOT familiar with the zap codebase). The python code code will pick only one (the last one), but earlier runs of the loop body may change prefix_chip_root_dir.  If I had removed prefix_chip_root_dir variable from the function altogether and replaced with an early return we will pick the first file instead changing the existing behavior so I chose not to.

Reply:
please use `prefix_chip_root_dir = (package["pathRelativity"] != "resolveEnvVars")` for clarity. Otherwise figuring out true/false requires parsing of nested if loops (and the corresponding if is not even the first one, it is the middle of several conditionals.

Reply:
ok, done